app/code/core/Maho/AdminActivityLog/Model/Observer.php

```diff
@@ -0,0 +1,521 @@
+        }
+    }
+
+    protected function _shouldLogActivity(): bool
+    {
+        return Mage::getStoreConfigFlag('admin/adminactivitylog/enabled')
+            && Mage::getSingleton('admin/session')->isLoggedIn();
+    }
+
+    protected function _isAdminArea(): bool
+    {
+        return Mage::app()->getStore()->isAdmin();
+    }
+
+    protected function _getCurrentModule(): string
+    {
+        return (string) Mage::app()->getRequest()->getModuleName();
+    }
+
+    protected function _getCurrentController(): string
+    {
+        return (string) Mage::app()->getRequest()->getControllerName();
+    }
+
+    protected function _getCurrentAction(): string
+    {
+        return (string) Mage::app()->getRequest()->getActionName();
+    }
+
+    protected function _getEntityType(Mage_Core_Model_Abstract $object): string
+    {
+        $class = $object::class;
+        $map = [
+            'Mage_Catalog_Model_Product' => 'product',
+            'Mage_Catalog_Model_Category' => 'category',
+            'Mage_Customer_Model_Customer' => 'customer',
+            'Mage_Sales_Model_Order' => 'order',
+            'Mage_Cms_Model_Page' => 'cms_page',
+            'Mage_Cms_Model_Block' => 'cms_block',
+            'Mage_Admin_Model_User' => 'admin_user',
+            'Mage_Admin_Model_Role' => 'admin_role',
+        ];
+
+        return $map[$class] ?? strtolower(str_replace('_Model_', '_', $class));
+    }
+
+    protected function _getEntityName(Mage_Core_Model_Abstract $object): string
+    {
+        $nameFields = ['name', 'title', 'sku', 'increment_id', 'username', 'email', 'identifier'];
+
+        foreach ($nameFields as $field) {
+            if ($object->hasData($field) && $object->getData($field)) {
+                return (string) $object->getData($field);
+            }
+        }
+
+        return 'ID: ' . $object->getId();
+    }
+
+    public function logMassAction(Varien_Event_Observer $observer): void
+    {
+        if (!Mage::getStoreConfigFlag('admin/adminactivitylog/enabled')) {
+            return;
+        }
+
+        if (!Mage::getStoreConfigFlag('admin/adminactivitylog/log_mass_actions')) {
+            return;
+        }
+
+        if (!Mage::getSingleton('admin/session')->isLoggedIn()) {
+            return;
+        }
+
+        $controllerAction = $observer->getEvent()->getControllerAction();
+        $request = $controllerAction->getRequest();
+
+        // Get all parameters for debugging
+        $allParams = $request->getParams();
+        $actionName = $request->getActionName();
+        $controllerName = $request->getControllerName();
+        $moduleName = $request->getModuleName();
+
+        // Debug log what we're seeing
+        Mage::log("Mass Action Debug - Action: {$actionName}, Controller: {$controllerName}, Module: {$moduleName}", null, 'adminactivitylog_debug.log');
+        Mage::log('Parameters: ' . print_r($allParams, true), null, 'adminactivitylog_debug.log');
```

Comment:
There's a few debug logs still. Should we remove this, or put it behind `Mage::getIsDeveloperMode()` ?

Reply:
nah, let's remove those, sorry